front/src/modules/ui/board/components/BoardOptionsDropdownContent.tsx

```diff
@@ -1,54 +1,159 @@
-import { useState } from 'react';
+import { useCallback, useRef, useState } from 'react';
 import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useRecoilState } from 'recoil';
+import { Key } from 'ts-key-enum';
+import { v4 } from 'uuid';
 
 import { DropdownMenuHeader } from '@/ui/dropdown/components/DropdownMenuHeader';
+import { DropdownMenuInput } from '@/ui/dropdown/components/DropdownMenuInput';
 import { DropdownMenuItem } from '@/ui/dropdown/components/DropdownMenuItem';
 import { StyledDropdownMenu } from '@/ui/dropdown/components/StyledDropdownMenu';
 import { StyledDropdownMenuItemsContainer } from '@/ui/dropdown/components/StyledDropdownMenuItemsContainer';
 import { StyledDropdownMenuSeparator } from '@/ui/dropdown/components/StyledDropdownMenuSeparator';
-import { IconChevronLeft } from '@/ui/icon';
+import { useDropdownButton } from '@/ui/dropdown/hooks/useDropdownButton';
+import {
+  IconChevronLeft,
+  IconChevronRight,
+  IconLayoutKanban,
+  IconPlus,
+  IconSettings,
+} from '@/ui/icon';
+import { useScopedHotkeys } from '@/ui/utilities/hotkey/hooks/useScopedHotkeys';
+import { HotkeyScope } from '@/ui/utilities/hotkey/types/HotkeyScope';
 
-type BoardOptionsDropdownMenu = 'options' | 'fields';
+import { boardColumnsState } from '../states/boardColumnsState';
+import type { BoardColumnDefinition } from '../types/BoardColumnDefinition';
+import { BoardOptionsDropdownKey } from '../types/BoardOptionsDropdownKey';
 
-export function BoardOptionsDropdownContent() {
+type BoardOptionsDropdownContentProps = {
+  customHotkeyScope: HotkeyScope;
+  onStageAdd?: (boardColumn: BoardColumnDefinition) => void;
+};
+
+const StyledIconSettings = styled(IconSettings)`
+  margin-right: ${({ theme }) => theme.spacing(1)};
+`;
+
+const StyledIconChevronRight = styled(IconChevronRight)`
+  color: ${({ theme }) => theme.font.color.tertiary};
+  margin-left: auto;
+`;
+
+enum BoardOptionsMenu {
+  StageCreation = 'StageCreation',
+  Stages = 'Stages',
+}
+
+export function BoardOptionsDropdownContent({
+  customHotkeyScope,
+  onStageAdd,
+}: BoardOptionsDropdownContentProps) {
   const theme = useTheme();
 
-  const [menuShown, setMenuShown] =
-    useState<BoardOptionsDropdownMenu>('options');
+  const stageInputRef = useRef<HTMLInputElement>(null);
+
+  const [currentMenu, setCurrentMenu] = useState<
+    BoardOptionsMenu | undefined
+  >();
+
+  const [boardColumns, setBoardColumns] = useRecoilState(boardColumnsState);
+
+  const resetMenu = useCallback(() => {
```

Comment:
Could we keep useCallback for compute intensive functions ? Wrapping every function in a useCallback is making the code less readable than it optimizes the compute, see this excellent article for reference : https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Done